src/meta/app/src/schema/catalog.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::ops::Deref;
+
+use common_datavalues::chrono::DateTime;
+use common_datavalues::chrono::Utc;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum CatalogType {
+    Default = 0,
+    Hive = 1,
+}
+
+impl Default for CatalogType {
+    fn default() -> Self {
+        CatalogType::Default
+    }
+}
+
+impl Display for CatalogType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CatalogType::Default => write!(f, "DEFAULT"),
+            CatalogType::Hive => write!(f, "HIVE"),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq)]
+pub struct CatalogNameIdent {
+    pub tenant: String,
+    pub ctl_name: String,
+}
+
+impl Display for CatalogNameIdent {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "'{}'/'{}'", self.tenant, self.ctl_name)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq)]
+pub struct CatalogInfo {
+    pub name_ident: CatalogNameIdent,
+    pub meta: CatalogMeta,
+}
+
+impl CatalogInfo {
+    pub fn catalog_type(&self) -> CatalogType {
+        self.meta.catalog_type
+    }
+}
```

Comment:
Remove this if this is just an analog of `TableInfo`. `TableInfo` is kept for only historical reasons.